records/generator.go

```diff
@@ -293,27 +296,32 @@ func (rg *RecordGenerator) InsertState(sj StateJSON, domain string, ns string,
 			tag := hashString(task.ID)
 			tail := fname + "." + domain + "."
 
-			// A records for task and task-sid
 			arec := tname + "." + tail
-			rg.insertRR(arec, hostIP, "A")
 			trec := tname + "-" + tag + "-" + sid + "." + tail
-			rg.insertRR(trec, hostIP, "A")
 
-			// A records with container IP
-			if containerIP := task.containerIP(); containerIP != "" {
-				rg.insertRR("_container."+arec, containerIP, "A")
-				rg.insertRR("_container."+trec, containerIP, "A")
+			containerIP := ""
+			if containerIPsOn {
+				containerIP = task.containerIP()
 			}
-
-			// SRV records
-			if task.Resources.Ports != "" {
-				ports := yankPorts(task.Resources.Ports)
-				for _, port := range ports {
-					srvhost := trec + ":" + port
-					tcp := "_" + tname + "._tcp." + tail
-					udp := "_" + tname + "._udp." + tail
-					rg.insertRR(tcp, srvhost, "SRV")
-					rg.insertRR(udp, srvhost, "SRV")
+			if containerIP != "" {
+				// A records with container IP, no SRV records here
+				rg.insertRR(arec, containerIP, "A")
+				rg.insertRR(trec, containerIP, "A")
+			} else {
```

Comment:
if we support A records for "my-task.my-framework.slave.mesos" then we need to nix this else condition and generate those records here.

we should also continue to generate SRV's here for any host ports allocated by the task, regardless of whether there's a container IP (why should mesos-dns make assumptions about such task's?)